src/server.rs

```diff
@@ -397,8 +397,14 @@ impl<O: Output> LsService<O> {
         self.output.success(id, ResponseData::Init(result));
 
         let root_path = params.root_path.map(PathBuf::from).expect("No root path");
+        let init_options: Option<InitializationOptions> = params.initialization_options
+                              .and_then(|options| serde_json::from_value(options)
```

Comment:
@nrc this is somewhat more user-facing and gives LSP clients freedom to pass any JSON config (which is reasonable, since the configs will wildly differ for different servers), hence why I wanted to have a debug or preferably a user warning later that passed config is invalid or unsupported by the current server. Should I simplify that for now?